### PR TITLE
Feature/aql modificators cpp test

### PR DESCRIPTION
**Pull request by mchacki**
This just adds more C++ test for all Modification Executors.
We did some refactoring there in a different branch and we want to make sure we have not broken anything.

No production changes just additional google test code.

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6803/

### CODE REVIEW

**Comment by markuspf** [comment]
It should be highlighted that this PR contains fixes for the test mocks (and which those are).

File: tests/Aql/InsertExecutorTest.cpp [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+      arangodb::tests::executeQuery(vocbase,
+                                    "FOR d IN " + collectionName + " RETURN d",
+                                    bindParameters);
+  ASSERT_TRUE(queryResult.ok());
+  auto slice = queryResult.data->slice();
+  ASSERT_TRUE(slice.isArray());
+  ASSERT_EQ(slice.length(), 1);
+
+  auto doc = slice.at(0);
+  ASSERT_NE(arangodb::basics::VelocyPackHelper::getStringValue(doc, "_rev", ""),
```
Comment (mchacki) [inline tests/Aql/InsertExecutorTest.cpp]: I think this test is not correct.
It tries to write "bar", and asserts that it has not written "foo", where I assume it wanted to make sure "bar" is not written by server generated.

File: tests/Aql/InsertExecutorTest.cpp [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+  builder.openArray();
+  for (size_t i = 1; i <= nDocs; i++) {
+    builder.add(VPackValue(i));
+  }
+  builder.close();
+
+  AssertQueryHasResult(vocbase, query, builder.slice());
+}
+
+TEST_P(InsertExecutorTestCount, insert_with_key_and_overwrite) {
```
Comment (mchacki) [inline tests/Aql/InsertExecutorTest.cpp]: This test is correct and good, however I wonder what is the difference between INSERT overwrite:true
and REPSERT?

File: tests/Aql/RemoveExecutorTest.cpp [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+TEST_P(RemoveExecutorTestPatterns, remove_all_without_return) {
+  std::string query =
+      std::string("FOR d IN " + collectionName + " REMOVE d IN " + collectionName);
+
+  AssertQueryHasResult(vocbase, query, VPackSlice::emptyArraySlice());
+
+  std::string checkQuery = "FOR i IN " + collectionName + " RETURN i.value";
+  AssertQueryHasResult(vocbase, checkQuery, VPackSlice::emptyArraySlice());
```
Comment (mchacki) [inline tests/Aql/RemoveExecutorTest.cpp]: Can you add an assertion that collection is now empty please

File: tests/Aql/RemoveExecutorTest.cpp [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  auto const bindParameters = VPackParser::fromJson("{ }");
+  std::string allQuery = std::string("FOR d IN " + collectionName + " RETURN d");
+
+  auto allDocs = executeQuery(vocbase, allQuery, bindParameters);
+  ASSERT_TRUE(allDocs.ok());
+
+  std::string query =
+      std::string("FOR d IN " + collectionName + " REMOVE d IN " +
+                  collectionName + " RETURN OLD");
+  AssertQueryHasResult(vocbase, query, allDocs.data->slice());
```
Comment (mchacki) [inline tests/Aql/RemoveExecutorTest.cpp]: Can you add an assertion that collection is now empty please

File: tests/Aql/InsertExecutorTest.cpp (hunk 2 above)
Comment (markuspf) [reply tests/Aql/InsertExecutorTest.cpp]: with `REPSERT` you can insert a different document for the `INSERT` case vs the `REPLACE` case, I guess.

Comment (markuspf) [reply tests/Aql/InsertExecutorTest.cpp]: with `REPSERT` you can insert a different document for the `INSERT` case vs the `REPLACE` case, I guess.

File: tests/Aql/UpsertExecutorTest.cpp [hunk 5]
```diff
@@ -0,0 +1,811 @@
...
+  TRI_vocbase_t& vocbase{server.getSystemDatabase()};
+
+  void SetUp() override {
+    SCOPED_TRACE("Setup");
+    auto info = VPackParser::fromJson(R"({"name":"UnitTestCollection"})");
+    auto collection = vocbase.createCollection(info->slice());
+    ASSERT_NE(collection.get(), nullptr) << "Failed to create collection";
+    // Insert Documents
+    std::string insertQuery =
+        R"aql(FOR i IN 1..)aql" + basics::StringUtils::itoa(numDocs()) +
```
[the remark this answers is not in the document]
Comment (mchacki) [reply tests/Aql/UpsertExecutorTest.cpp]: I think it is pre c++11 knowledge...
The underlying function is rather, non straight forward ;)
Will take std::to_string.

**Comment by mchacki** [comment]
Once again,
after comment merges:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/

**Comment by mchacki** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6912/

**Comment by markuspf** [comment]
The non maintainer compile failed because c05 seems to have been broken. I started yet another compile to be sure here: http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6922/

**Comment by markuspf** [comment]
Tests are blue

**Pull request merged**